lib/liveValidation/operationSearcher.ts

```diff
@@ -343,11 +343,22 @@ export function getProviderFromPathTemplate(pathStr?: string | null): string | u
 }
 const providerRegEx = new RegExp("/providers/(:?[^{/]+)", "gi");
 
-export function getProviderFromSpecPath(specPath: string): string | undefined {
-  const match = providerInSpecPathRegEx.exec(specPath);
-  return match === null ? undefined : match[1];
+export interface PathProvider {
+  provider: string;
+  type: "resource-manager" | "data-plane";
 }
-const providerInSpecPathRegEx = new RegExp("/resource-manager/(:?[^{/]+)", "gi");
+
+export function getProviderFromSpecPath(specPath: string): PathProvider | undefined {
```

Comment:
@zhenglaizhang this looks like a place where we make some assumptions about the path structure and conventions in the specs repo. Do we know how many places we are doing this similar processing across all our various tools? We probably need to extract this logic out into one tool or script so that we can avoid this duplication.

Reply:
Thanks @weshaggard for the suggestion. There're a few places describing spec repo path pattern but not all of them are same. We will need to collect them and revisit whether they could be consolidated.